src/shell/mod.rs

```diff
@@ -1782,6 +1782,8 @@ impl Shell {
         };
     }
 
+    /// Determine whether a Surface that has changed should be focused, and if so, return the target. Only called when the
+    /// Compositor is handling visibile state changes.
```

Comment:
This isn't entirely correct. This method does "map" windows somewhere onto a workspace/output. Otherwise they aren't visible, as they are not part of the internal tree representation of the shell.
This can effect focus, which is why it might return a new focus target, but it isn't concerned with only focus.